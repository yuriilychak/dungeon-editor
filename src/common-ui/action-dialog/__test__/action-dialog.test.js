import React from "react";
import ActionDialog from "../action-dialog";
import { createMount } from '@material-ui/core/test-utils';

describe('ActionDialog test',()=>{

    const title = "Test";
    const description = "Test";
    const dialogId = "test";
    const onSubmit = jest.fn();
    const onReject = jest.fn();

    const wrapper = createMount()(
        <ActionDialog
            open={true}
            title={title}
            description={description}
            dialogId={dialogId}
            onSubmit={onSubmit}
            onReject={onReject}
            submitTitle={title}
            rejectTitle={title}
        />
    );

    const projectInput = wrapper.find('input');

    it('Open snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Close snapshot', () => {
        const wrapper = createMount()(
            <ActionDialog
                open={true}
                title={title}
                description={description}
                dialogId={dialogId}
                onSubmit={onSubmit}
                onReject={onReject}
                submitTitle={title}
                rejectTitle={title}
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Click submit', () => {
        const button = wrapper.find('#test-submit').first();
        button.simulate('click');
        expect(onSubmit).toHaveBeenCalled();
    });

    it('Click reject', () => {
        const button = wrapper.find('#test-reject').first();
        button.simulate('click');
        expect(onReject).toHaveBeenCalled();
    });
});
