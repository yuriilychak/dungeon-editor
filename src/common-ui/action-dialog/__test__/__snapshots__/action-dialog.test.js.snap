// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionDialog test Close snapshot 1`] = `"<div role=\\"dialog\\" class=\\"MuiModal-root-15 MuiDialog-root-1\\" aria-labelledby=\\"test-title\\"><div class=\\"MuiBackdrop-root-17\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"MuiDialog-container-4 MuiDialog-scrollPaper-2\\" role=\\"document\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root-19 MuiPaper-elevation24-45 MuiPaper-rounded-20 MuiDialog-paper-5 action-dialog-container MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9\\"><div class=\\"MuiDialogTitle-root-46\\" id=\\"test-title\\"><h2 class=\\"MuiTypography-root-47 MuiTypography-title-53\\">Test</h2></div><div class=\\"MuiDialogContent-root-83\\"><p class=\\"MuiTypography-root-47 MuiDialogContentText-root-84 MuiTypography-subheading-54 MuiTypography-colorTextSecondary-80\\">Test</p></div><div class=\\"MuiDialogActions-root-85\\"><button class=\\"MuiButtonBase-root-113 MuiButton-root-87 MuiButton-contained-98 MuiButton-containedSecondary-100 MuiButton-raised-101 MuiButton-raisedSecondary-103 action-dialog-button MuiDialogActions-action-86\\" tabindex=\\"0\\" type=\\"button\\" id=\\"test-reject\\"><span class=\\"MuiButton-label-88\\">Test</span><span class=\\"MuiTouchRipple-root-116\\"></span></button><button class=\\"MuiButtonBase-root-113 MuiButton-root-87 MuiButton-contained-98 MuiButton-containedPrimary-99 MuiButton-raised-101 MuiButton-raisedPrimary-102 action-dialog-button MuiDialogActions-action-86\\" tabindex=\\"0\\" type=\\"button\\" id=\\"test-submit\\"><span class=\\"MuiButton-label-88\\">Test</span><span class=\\"MuiTouchRipple-root-116\\"></span></button></div></div></div></div>"`;

exports[`ActionDialog test Open snapshot 1`] = `"<div role=\\"dialog\\" class=\\"MuiModal-root-15 MuiDialog-root-1\\" aria-labelledby=\\"test-title\\"><div class=\\"MuiBackdrop-root-17\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"MuiDialog-container-4 MuiDialog-scrollPaper-2\\" role=\\"document\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root-19 MuiPaper-elevation24-45 MuiPaper-rounded-20 MuiDialog-paper-5 action-dialog-container MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9\\"><div class=\\"MuiDialogTitle-root-46\\" id=\\"test-title\\"><h2 class=\\"MuiTypography-root-47 MuiTypography-title-53\\">Test</h2></div><div class=\\"MuiDialogContent-root-83\\"><p class=\\"MuiTypography-root-47 MuiDialogContentText-root-84 MuiTypography-subheading-54 MuiTypography-colorTextSecondary-80\\">Test</p></div><div class=\\"MuiDialogActions-root-85\\"><button class=\\"MuiButtonBase-root-113 MuiButton-root-87 MuiButton-contained-98 MuiButton-containedSecondary-100 MuiButton-raised-101 MuiButton-raisedSecondary-103 action-dialog-button MuiDialogActions-action-86\\" tabindex=\\"0\\" type=\\"button\\" id=\\"test-reject\\"><span class=\\"MuiButton-label-88\\">Test</span><span class=\\"MuiTouchRipple-root-116\\"></span></button><button class=\\"MuiButtonBase-root-113 MuiButton-root-87 MuiButton-contained-98 MuiButton-containedPrimary-99 MuiButton-raised-101 MuiButton-raisedPrimary-102 action-dialog-button MuiDialogActions-action-86\\" tabindex=\\"0\\" type=\\"button\\" id=\\"test-submit\\"><span class=\\"MuiButton-label-88\\">Test</span><span class=\\"MuiTouchRipple-root-116\\"></span></button></div></div></div></div>"`;
